vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/ExtendedQueryCommandBaseCodec.java

```diff
@@ -44,158 +36,76 @@ void encode(TdsMessageEncoder encoder) {
-    while (messageBody.isReadable()) {
-      int tokenType = messageBody.readUnsignedByte();
-      MSSQLPreparedStatement ps = (MSSQLPreparedStatement) cmd.preparedStatement();
-      switch (tokenType) {
-        case COLMETADATA:
-          MSSQLRowDesc rowDesc = decodeColmetadataToken(messageBody);
-          rowResultDecoder = new RowResultDecoder<>(cmd.collector(), rowDesc);
-          break;
-        case ROW:
-          handleRow(messageBody);
-          break;
-        case NBCROW:
-          handleNbcRow(messageBody);
-          break;
-        case DONE:
-          messageBody.skipBytes(12); // this should only be after ERROR?
-          break;
-        case INFO:
-        case ORDER:
-          int tokenLength = messageBody.readUnsignedShortLE();
-          messageBody.skipBytes(tokenLength);
-          break;
-        case ERROR:
-          handleErrorToken(messageBody);
-          break;
-        case DONEPROC:
-          messageBody.skipBytes(12);
-          handleResultSetDone(rowCount);
-          break;
-        case DONEINPROC:
-          short status = messageBody.readShortLE();
-          short curCmd = messageBody.readShortLE();
-          long doneRowCount = messageBody.readLongLE();
-          if ((status | DoneToken.STATUS_DONE_FINAL) != 0) {
-            rowCount += doneRowCount;
-          } else {
-            handleResultSetDone((int) doneRowCount);
-          }
-          break;
-        case RETURNSTATUS:
-          messageBody.skipBytes(4);
-          break;
-        case RETURNVALUE:
-          if (ps.handle == 0) {
-            messageBody.skipBytes(2); // skip ordinal position
-            messageBody.skipBytes(2 * messageBody.readUnsignedByte()); // skip param name
-            messageBody.skipBytes(1); // skip status
-            messageBody.skipBytes(4); // skip user type
-            messageBody.skipBytes(2); // skip flags
-            messageBody.skipBytes(1); // skip type id
-            messageBody.skipBytes(2); // max length and length
-            ps.handle = messageBody.readIntLE();
-          }
-          messageBody.skipBytes(messageBody.readableBytes()); // FIXME
-          handleResultSetDone(rowCount);
-          break;
-        default:
-          throw new UnsupportedOperationException("Unsupported token: " + tokenType);
-      }
+  protected void handleDone(short tokenType) {
+    if (tokenType == TokenType.DONEPROC) {
+      handleResultSetDone();
     }
-    handleMessageDecoded();
   }
 
   @Override
-  protected void handleResultSetDone(int affectedRows) {
-    super.handleResultSetDone(rowCount);
+  protected void handleResultSetDone() {
+    super.handleResultSetDone();
     rowCount = 0;
   }
 
-  protected abstract void handleMessageDecoded();
+  @Override
+  protected void handleReturnValue(ByteBuf payload) {
+    MSSQLPreparedStatement ps = (MSSQLPreparedStatement) cmd.preparedStatement();
+    if (ps.handle == 0) {
+      payload.skipBytes(2); // skip ordinal position
```

Comment:
consider calling this a single time, it looks cute but in practice it costs (it falls in the same area as `write`)

Reply:
e.g you could do something like

```java
int bytesToSkip = 2 // sky ordinal position
                           + 2 * payload.readUnsignedByte() // skip param name
                           ...;
```

Reply:
Done